mithril-client/src/snapshot_client.rs

```diff
@@ -188,56 +209,161 @@ impl SnapshotClient {
+            self.download_unpack_immutables_files(snapshot, target_dir, &download_id)
                 .await?;
+
+            Ok(())
+        }
+
+        async fn download_unpack_immutables_files(
+            &self,
+            snapshot: &Snapshot,
+            target_dir: &std::path::Path,
+            download_id: &str,
+        ) -> MithrilResult<()> {
+            self.download_unpack_file(
+                &snapshot.digest,
+                &snapshot.locations,
+                snapshot.size,
+                target_dir,
+                snapshot.compression_algorithm,
+                DownloadEvent::Full {
+                    download_id: download_id.to_string(),
+                    digest: snapshot.digest.clone(),
+                },
+            )
+            .await?;
+
+            Ok(())
+        }
+
+        async fn download_unpack_ancillary(
+            &self,
+            snapshot: &Snapshot,
+            target_dir: &std::path::Path,
+            download_id: &str,
+        ) -> MithrilResult<()> {
+            slog::info!(self.logger, "Ancillary verification don't use Mithril certification");
+
+            match &snapshot.ancillary_locations {
+                None => Ok(()),
+                Some(ancillary_locations) => {
+                    let temp_ancillary_unpack_dir = Self::ancillary_subdir(target_dir, download_id);
+                    tokio::fs::create_dir(&temp_ancillary_unpack_dir)
+                        .await
+                        .with_context(|| {
+                            format!(
+                                "Snapshot Client can not create ancillary unpack directory '{}'",
+                                temp_ancillary_unpack_dir.display()
+                            )
+                        })?;
+
+                    let result = self
+                        .download_unpack_verify_ancillary(
+                            snapshot,
+                            ancillary_locations,
+                            snapshot.ancillary_size.unwrap_or(0),
+                            target_dir,
+                            &temp_ancillary_unpack_dir,
+                            download_id,
+                        )
+                        .await;
+
+                    if let Err(e) = std::fs::remove_dir_all(&temp_ancillary_unpack_dir) {
+                        slog::warn!(
+                            self.logger, "Failed to remove ancillary unpack directory '{}'", temp_ancillary_unpack_dir.display();
+                            "error" => ?e
+                        );
+                    }
+
+                    result
+                }
             }
+        }
+
+        async fn download_unpack_verify_ancillary(
+            &self,
+            snapshot: &Snapshot,
+            ancillary_locations: &[String],
+            ancillary_size: u64,
+            target_dir: &std::path::Path,
+            temp_ancillary_unpack_dir: &std::path::Path,
```

Comment:
There are a lot of repetition of `std::path::Path` in this module which limits the readability of the function signatures